ONSdigital/census-contact-centre-cucumber
# Cr 771 - CC Automation test POSTAL fulfilments

# Motivation and Context

This change is required in order to verify that the getFulfilments endpoint, in the CCSVC, returns the
correct fulfilment codes and sends the correct fulfilment requests to RM. The following 9 scenarios are 
tested but some of them throw a PENDING exception as the required products are not currently listed in
the product reference library:
1. [CR-T142] I want to request an UAC for a HH Respondent in NI via POST
2. [CR-T292] PENDING I want to request an UAC for a HI Respondent in Wales in welsh via Post
3. [CR-T301] I want to request a Welsh Paper Questionnaire for a CE Individual Respondent in Wales
4. [CR-T302] PENDING I want to request an UAC for a CE Individual Respondent in Wales in welsh via Post
5. [CR-T304] PENDING I want to request a welsh Paper Questionnaire for a CE Manager in Wales
6. [CR-T313] PENDING I want to request an UAC for a CE Individual Respondent in NI via Post
7. [CR-T316] PENDING I want to request an UAC for a CE Manager in NI via Post
8. [CR-T323] I want to request a Paper Questionnaire for a SPG Individual Respondent in NI
9. [CR-T334] PENDING I want to request an UAC for a SPG Respondent in Wales in welsh language via Post

When the product reference library has been updated then the word 'PENDING' will need to be removed
from the relevant scenario names but the code steps should automatically locate the correct products 
and therefore should not require any changes (because the exception is thrown only if no matching
product is found in the product reference libary).

# What has changed

The testFulfilmentsEndpoint feature has had the 9 scnearios added to it.

# How to test?

Run the contact centre cucumber tests locally by doing as follows:
- run rabbitmq (e.g. go to census-rh-service repo and use this command: docker-compose up -d)
- run the latest version of the mock case service
- run the latest version of the contact centre service
- go to the census-contact-centre-cucumber repo and run the tests using this command: mvn clean install

The results for the testFulfilmentsEndpoint feature should look like this:

Pending scenarios:
src/test/resources/integrationtests/fulfilments/testFulfilmentsEndpoint.feature:62 # [CR-T292] PENDING I want to request an UAC for a HI Respondent in Wales in welsh via Post
src/test/resources/integrationtests/fulfilments/testFulfilmentsEndpoint.feature:82 # [CR-T302] PENDING I want to request an UAC for a CE Individual Respondent in Wales in welsh via Post
src/test/resources/integrationtests/fulfilments/testFulfilmentsEndpoint.feature:93 # [CR-T304] PENDING I want to request a welsh Paper Questionnaire for a CE Manager in Wales
src/test/resources/integrationtests/fulfilments/testFulfilmentsEndpoint.feature:104 # [CR-T313] PENDING I want to request an UAC for a CE Individual Respondent in NI via Post
src/test/resources/integrationtests/fulfilments/testFulfilmentsEndpoint.feature:115 # [CR-T316] PENDING I want to request an UAC for a CE Manager in NI via Post
src/test/resources/integrationtests/fulfilments/testFulfilmentsEndpoint.feature:135 # [CR-T334] PENDING I want to request an UAC for a SPG Respondent in Wales in welsh language via Post

29 Scenarios (6 pending, 23 passed)
108 Steps (6 skipped, 6 pending, 96 passed)

NB. This PR cannot be merged until the following PR has also been merged and the CCSVC updated to point at the new version of the CCSVC API:
https://github.com/ONSdigital/census-contact-centre-service-api
